### PR TITLE
Update Runtime APIs 

### PR DESCRIPTION
**Pull request by bhargavbh**
Addresses issues raised in https://github.com/w3f/polkadot-spec/issues/604 and https://github.com/w3f/polkadot-spec/issues/605.
Adds spec for APIs regarding ParachainHost, TransacationPayment, NominationPools, and Grandpa.
Does not include Beefy and MMR related API as the spec for these chapters is WIP, makes sense to include them only after key definitions have been added in Spec .

### CODE REVIEW

File: ac_runtime-api/modules/parachain.adoc [hunk 1]
```diff
@@ -305,3 +332,83 @@ Arguments::
...
+
+==== `ParachainHost_on_chain_votes`
+
+Returns disputes relevant from on-chain, backing votes, and resolved disputes. 
+
+Arguments::
+* None
+
+Return::
+* An _Option_ type which can contain the Scraped OnChainVotes data (<<defn-scraped-on-chain-vote>>).
```
Comment (lamafab) [inline ac_runtime-api/modules/parachain.adoc]: * `"An _Option_  value (<<defn-option-type>>) ..."`
* Custom (Rust) type names should be italic, i.e. `"_OnChainVotes_"`. Alternatively (or preferably), just say `"... can contain the scraped on-chain votes data..."`.

File: ac_runtime-api/modules/parachain.adoc [hunk 2]
```diff
@@ -305,3 +332,83 @@ Arguments::
...
+
+Return::
+* An _Option_ type which can contain the Scraped OnChainVotes data (<<defn-scraped-on-chain-vote>>). 
+
+[#defn-scraped-on-chain-vote]
+.<<defn-scraped-on-chain-vote, Scraped On Chain Vote>>
+====
+Contains the scraped runtime backing votes and resolved disputes.
+
+The ScrapedOnChainVotes, stem:[SOCV], is a datastructure of the following format:
```
Comment (lamafab) [inline ac_runtime-api/modules/parachain.adoc]: Same here, do italic or just `"The scraped on-chain votes ..."`

File: ac_runtime-api/modules/nominationpools.adoc [hunk 3]
```diff
@@ -0,0 +1,16 @@
+[#sect-runtime-nomination-pools-module]
+=== Module Nomination Pools
+
+IMPORTANT: This section describes *Version 1* of this API. Please check `Core_version` (<<defn-rt-core-version>>) to ensure compatibility. Currently supports only one RPC endpoint. 
+
+
+[#sect-rte-pending-rewards]
+==== `NominationPools_pending_rewards`
```
Comment (lamafab) [inline ac_runtime-api/modules/nominationpools.adoc]: * The `NominationPools` prefix should be `NominationPoolsApi`.
* It looks like the `points_to_balance` and `balance_to_points` APIs are missing?

File: ac_runtime-api/modules/parachain.adoc [hunk 4]
```diff
@@ -305,3 +332,83 @@ Arguments::
...
+* A PVF pre checking statement (<<defn-pvf-check-statement>>) to be submitted into the transaction pool.
+* Validator Signature (<<defn-parachain-inherent-data>>).
+
+Return::
+* None
+
+[#defn-pvf-check-statement]
+.<<defn-pvf-check-statement, PVF Check Statement>>
+====
+This is a statement by the validator who ran the pre-checking process for a PVF. A PVF is identified by the `ValidationCodeHash`. The statement is valid only during a single session, specified in the `session_index`.
```
Comment (lamafab) [inline ac_runtime-api/modules/parachain.adoc]: Same here with `"_ValidationCodeHash_"`

File: ac_runtime-api/modules/parachain.adoc [hunk 5]
```diff
@@ -305,3 +332,83 @@ Arguments::
...
+
+[stem]
+++++
+SOCV = (S_i,BV,d)\
+BV = [C_r, [(i,a)]]
+++++
+
+where::
+* stem:[S_i] is the u32 integer representing the session index in which block was introduced.
+* stem:[BV] is the set of backing validators for each candidate, represented by its candidate receipt (<<defn-candidate-receipt>>). Each candidate stem:[C_r] has the list [stem:[(i,a)]], the pair of validator index and validation attestations (<<defn-parachain-inherent-data>>).
```
Comment (lamafab) [inline ac_runtime-api/modules/parachain.adoc]: > list [stem:[(i,a)]]

formatting mistake?

File: ac_runtime-api/modules/txpayment.adoc [hunk 6]
```diff
@@ -73,3 +75,67 @@ where::::
 *** latexmath:[f_l] is the length fee, the amount paid for the encoded length (in bytes) of the extrinsic.
```
Comment (lamafab) [inline ac_runtime-api/modules/txpayment.adoc:75]: > where::::
> *** ...

Is the formatting correct here?

File: ac_runtime-api/modules/txpayment.adoc [hunk 7]
```diff
@@ -73,3 +75,67 @@ where::::
...
+
+Return::
+* A data structure of the following format:
++
+[latexmath]
+++++
+(w, c, f)
+++++
++
+where:::
```
Comment (lamafab) [inline ac_runtime-api/modules/txpayment.adoc:96]: > where:::

Same here

File: ac_runtime-api/modules/txpayment.adoc [hunk 8]
```diff
@@ -73,3 +75,67 @@ where::::
...
 *** latexmath:[f_a] is the "`adjusted weight fee`", which is a multiplication of the fee multiplier and the weight fee. The fee multiplier varies depending on the usage of the network.
 ** latexmath:[t] is the tip for the block author.
+
+[#sect-rte-transactionpaymentapi-query-call-info]
+==== `TransactionPaymentApi_query_call_info`
+
+Query information of a dispatch class, weight, and fee of a given encoded `Call`.
+
+Arguments::
+* A byte array of varying sizes containing the `Call`.
```
Comment (lamafab) [inline ac_runtime-api/modules/txpayment.adoc:85]: We always avoided using overly Rust/Substrate specific concepts like `Call`. Simply refer to it as "extrinsic" (and for all other mentions of `Call`).

**Review by lamafab — CHANGES_REQUESTED** [review]
Quick feedback

File: ac_runtime-api/modules/txpayment.adoc (hunk 6 above)
Comment (bhargavbh) [reply ac_runtime-api/modules/txpayment.adoc:75]: This was from a previous commit. It is correct, it lets you nest deeper and provides the right indentation. removing it flattens the indentation

File: ac_runtime-api/modules/txpayment.adoc (hunk 7 above)
Comment (bhargavbh) [reply ac_runtime-api/modules/txpayment.adoc:96]: same as above

File: ac_runtime-api/modules/nominationpools.adoc (hunk 3 above)
Comment (bhargavbh) [reply ac_runtime-api/modules/nominationpools.adoc]: Fixed the first type. Regarding second, `points_to_balance` and `balance_to_points` are not declared as Runtime APIs.

File: ac_runtime-api/modules/txpayment.adoc (hunk 8 above)
Comment (bhargavbh) [reply ac_runtime-api/modules/txpayment.adoc:85]: There seems to be a difference between a runtime call and an extrinsic. Could you please provide any link where this is evident? the `transactionpaymentapi_query_info` and `transactionpaymentapi_query_fee_details` already exist for extrinsics. The new APIs `transactionpaymentapi_query_call_info` and `transactionpaymentapi_query_call_fee_details` are separately declared, however the type of RuntimeCall and Extrinsic seem to be the same from substrate code. We may need to add a definition for `Runtime Call` and explain how they are different from Extrinsics.

**Review by Noc2 — APPROVED** [review]

**Comment by 0xCaso** [comment]
Closed to finish it using the new format here: https://github.com/w3f/polkadot-spec/pull/643

**Pull request closed without merging**